### PR TITLE
Disambiguate overloads of Machine::is_self.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

File: iocore/utils/I_Machine.h [hunk 1]
```diff
@@ -77,6 +77,11 @@ struct Machine {
   /// @return The global instance of this class.
   static self_type *instance();
   bool is_self(std::string_view name);
+  bool
+  is_self(char const *name)
+  {
+    return this->is_self(std::string_view(name));
+  }
```
Comment (bneradt) [inline iocore/utils/I_Machine.h:84]: Probably best to keep the definition in `iocore/utils/Machine.cc` like the rest of the functions here.

**Review by bneradt — CHANGES_REQUESTED** [review]
Thanks for fixing this Walt!

File: iocore/utils/I_Machine.h (hunk 1 above)
Comment (ywkaras) [reply iocore/utils/I_Machine.h:84]: OK

Comment (ywkaras) [reply iocore/utils/I_Machine.h:84]: Not good for optimal inlining, but not called in any high-rep loop I don't think.

**Review by bneradt — COMMENTED** [review]
Looks good to me.

**Comment by ywkaras** [comment]
> Looks good to me.

You've still got it blocked.

**Review by randall — APPROVED** [review]

**Review by bneradt — APPROVED** [review]
Thanks Walt!

**Pull request merged**